price-feeder/oracle/client/client.go

```diff
@@ -93,7 +93,17 @@ func NewOracleClient(
 		return OracleClient{}, err
 	}
 
-	chainHeight, err := NewChainHeight(context.Background(), clientCtx.Client, oracleClient.Logger)
+	blockHeight, err := rpc.GetChainHeight(clientCtx)
+	if err != nil {
+		return OracleClient{}, err
+	}
+
+	chainHeight, err := NewChainHeight(
+		context.Background(),
```

Comment:
We need some top down context, rather than using Background in the middle of the flow. We can handle it later.

Reply:
I'll put it in a follow-up PR